code/game/jobs/job/job_exp.dm

```diff
@@ -240,4 +240,18 @@ GLOBAL_PROTECT(exp_to_update)
 			"ckey" = "'[sanitizeSQL(ckey)]'",
 			"minutes" = jvalue)))
 		prefs.exp[jtype] += jvalue
-	addtimer(CALLBACK(GLOBAL_PROC, /proc/update_exp_db), 20, TIMER_OVERRIDE|TIMER_UNIQUE)
\ No newline at end of file
+	addtimer(CALLBACK(GLOBAL_PROC, /proc/update_exp_db), 20, TIMER_OVERRIDE|TIMER_UNIQUE)
+
+
+/proc/queen_age_check(client/C)
+	if(!istype(C) || !C.prefs?.exp)
```

Comment:
This proc is being called by passing a client. The sanity check shouldn't be necessary.